### PR TITLE
feat(atomic): Add documentation to search box

### PR DESCRIPTION
**Pull request by nrawji**
Updated the controller to include better documentation, removed slots, fixed a couple of (non-shadow dom related) accessibility issues and move duplicate accessibility attributes to the `combobox.ts` file.

### CODE REVIEW

File: packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx [hunk 1]
```diff
@@ -67,6 +71,7 @@ export class AtomicSearchBox implements AtomicSearchBoxOptions {
   constructor() {
     this.combobox = new Combobox({
       id: this._id,
+      strings: this.strings,
```
Comment (nrawji) [inline packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx:74]: I wasn't sure what the best way to access i18n inside a non-component class was. I'm not sure if this is the best practice @ThibodeauJF?

Comment (ThibodeauJF) [reply packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx:74]: Not sure, but if it works correctly it should be a good way to do it.

in the index.html script try this:
```
setTimeout(() => (searchInterface.language = 'fr'), 1000);
```

if the language change re-renders the strings correctly, it's good, if it doesn't, we'll have to find another way

File: packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx [hunk 2]
```diff
@@ -15,10 +15,12 @@ import SearchIcon from 'coveo-styleguide/resources/icons/svg/search.svg';
 export interface AtomicSearchBoxOptions {
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx:15]: I think we can remove this interface, document the props directly, and their default should be picked up by Stencil correctly

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/src/locales.json [hunk 3]
```diff
@@ -3,6 +3,14 @@
     "en": "Search",
     "fr": "Recherche"
   },
+  "searchBox": {
+    "en": "Insert a query. Press enter to send.",
+    "fr": ""
```
Comment (ThibodeauJF) [inline packages/atomic/src/locales.json:8]: I think if you look in the JSUI string file, you have the french for that:
`Insérer une requête. Appuyez sur Entrée pour envoyer.

**Review by olamothe — APPROVED** [review]

**Review by samisayegh — APPROVED** [review]

**Pull request merged**